src/rrules/performance_patches.jl

```diff
@@ -100,28 +100,39 @@ function Mooncake.rrule!!(
     end
     return out, _kron!_pb!!
 end
+
 @mooncake_overlay function LinearAlgebra._kron!(
-    out::Matrix{T}, a::Vector{T}, b::Matrix{T}
-) where {T<:IEEEFloat}
-    return LinearAlgebra._kron!(out, reshape(a, :, 1), b)::Matrix{T}
+    out::Tm, a::Tv, b::Tm
+) where {T<:IEEEFloat,Tm<:AbstractMatrix{T},Tv<:AbstractVector{T}}
+    return LinearAlgebra._kron!(out, reshape(a, :, 1), b)::Tm
 end
 @mooncake_overlay function LinearAlgebra._kron!(
-    out::Matrix{T}, a::Matrix{T}, b::Vector{T}
-) where {T<:IEEEFloat}
-    return LinearAlgebra._kron!(out, a, reshape(b, :, 1))::Matrix{T}
+    out::Tm, a::Tm, b::Tv
+) where {T<:IEEEFloat,Tm<:AbstractMatrix{T},Tv<:AbstractVector{T}}
+    return LinearAlgebra._kron!(out, a, reshape(b, :, 1))::Tm
+end
+
+function arrayify(
+    x::Tx, dx::TangentOrFData
+) where {
+    T<:IEEEFloat,Tx<:Union{LowerTriangular{T,Matrix{<:T}},UpperTriangular{T,Matrix{<:T}}}
+}
+    arrayify(x.data, _fields(dx).data)
```

Comment:
maybe something like
```suggestion
    _, ddata = arrayify(x.data, _fields(dx).data)                                                         
    return x, ddata
```
so the primal type is preserved?

Reply:
also maybe move this to `blas.jl`?

Reply:
See, [5212fb8](https://github.com/chalk-lab/Mooncake.jl/pull/891/commits/5212fb8c25bb8fc7b8114856c24bcec9fbe2b638) for an improved version which is safer if `x.data`'s non-active elements are not properly initialised to zero.